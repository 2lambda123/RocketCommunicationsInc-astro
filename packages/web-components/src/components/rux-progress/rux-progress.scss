:host {
    position: relative;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: var(--spacing-0);
    height: calc(var(--spacing-4) + var(--spacing-1)); //20px;
}

:host([hidden]) {
    display: none;
}

progress:after {
    content: '';
    position: absolute;
    border: 1px solid hotpink;
    width: 100%;
    height: 100%;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
}
.rux-progress[value] {
    appearance: none;
    background-color: var(--color-background-surface-default);
    border: 1px solid var(--color-background-surface-default);
    border-radius: var(--progress-radius-outer);
    box-shadow: 0 0 0 1px var(--color-background-interactive-default);
    height: calc(var(--spacing-4) + var(--spacing-050)); //20px;
    width: 100%;
    box-sizing: border-box;
    overflow: hidden;

    &::-webkit-progress-bar {
        background-color: transparent;
    }
    &::-webkit-progress-value {
        border-radius: 0 var(--spacing-4) var(--spacing-4) 0;
        height: var(--spacing-4); //16
        max-width: calc(100% - var(--spacing-050)); //respect margin spacing
        background: var(--color-background-interactive-default);
        transition: width 0.3s ease;
        -moz-transition: width 0.3s ease;
        -webkit-transition: width 0.3s ease;
        -o-transition: width 0.3s ease;
    }
    &::-ms-fill {
        border-radius: 0 var(--spacing-4) var(--spacing-4) 0;
        border: none;
        height: var(--spacing-4); //16
        max-width: calc(100% - var(--spacing-050));
        background-color: var(--color-background-interactive-default);
    }
    &::-moz-progress-bar {
        border-radius: 0 var(--spacing-4) var(--spacing-4) 0;
        height: var(--spacing-4); //16
        max-width: calc(100% - var(--spacing-050));
        background: var(--color-background-interactive-default);
    }
}
.rux-progress__value {
    font-family: var(--font-body-1-font-family);
    font-weight: var(--font-body-1-font-weight);
    font-size: var(--font-body-1-font-size);
    letter-spacing: var(--font-body-1-letter-spacing);
    line-height: var(--font-body-1-line-height);
    margin-left: var(--spacing-2);
    text-align: right;
    color: var(--color-text-primary);
}

@keyframes spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
