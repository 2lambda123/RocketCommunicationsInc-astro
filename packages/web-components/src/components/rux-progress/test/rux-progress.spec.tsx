import { newSpecPage } from '@stencil/core/testing'
import { RuxProgress } from '../rux-progress'

describe('rux-progress', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxProgress],
            html: `<rux-progress></rux-progress>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    // it('returns progress as string', async () => {
    //     const progress = new RuxProgress()
    //     progress.value = 10
    //     const string = progress._getProgressAsString()
    //     expect(string).toBe('10%')
    // })

    // it('returns zero percent if no value is given', async () => {
    //     const progress = new RuxProgress()
    //     const string = progress._getProgressAsString()
    //     expect(string).toBe('0%')
    // })
    // it('changes max to equal value if given value is greater than given max', async () => {
    //     const progress = new RuxProgress()
    //     progress._checkValueNotOverMax(100, 150)
    //     expect(progress.max).toBe(150)
    // })
    // it('does not modify max if max is greater than given value', async () => {
    //     const progress = new RuxProgress()
    //     progress._checkValueNotOverMax(100, 50)
    //     expect(progress.max).toBe(100)
    // })
    // it('has correct default values for max and value if none are given', async () => {
    //     const progress = new RuxProgress()
    //     expect(progress.max).toBe(100)
    //     expect(progress.value).toBe(undefined)
    // })
    // it('renders a progress bar of 0/100 if value provided is 0', async () => {
    //     const page = await newSpecPage({
    //         components: [RuxProgress],
    //         html: `<rux-progress value="0"></rux-progress>`,
    //     })
    //     expect(page.root).toMatchSnapshot()
    // })

    // it('only renders the value if max is set to ""', async () => {
    //     const page = await newSpecPage({
    //         components: [RuxProgress],
    //         html: `<rux-progress value="20" max="100"></rux-progress>`,
    //     })
    //     await page?.root?.setAttribute('max', '')
    //     await page.waitForChanges()
    //     expect(page.root).toMatchSnapshot()
    // })
})
