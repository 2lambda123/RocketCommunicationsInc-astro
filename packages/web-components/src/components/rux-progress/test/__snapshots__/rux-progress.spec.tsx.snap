// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-progress only renders the value if max is set to "" 1`] = `
<rux-progress max="" value="20">
  <mock:shadow-root>
    <progress class="rux-progress" max="NaN" part="progress" value="20"></progress><output class="rux-progress__value" part="output">20</output>
    <slot></slot>
  </mock:shadow-root>
</rux-progress>
`;

exports[`rux-progress renders 1`] = `
<rux-progress>
  <mock:shadow-root>
    <rux-indeterminate-progress></rux-indeterminate-progress>
    <slot></slot>
  </mock:shadow-root>
</rux-progress>
`;

exports[`rux-progress renders a progress bar of 0/100 if value provided is 0 1`] = `
<rux-progress value="0">
  <mock:shadow-root>
    <progress class="rux-progress" max="100" part="progress" value="0"></progress><output class="rux-progress__value" part="output">0%</output>
    <slot></slot>
  </mock:shadow-root>
</rux-progress>
`;
